src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -1300,22 +1309,75 @@ namespace DataZoneEquipment {
 		ControlledZoneIndex = FindItemInList( ZoneName, ZoneEquipConfig, &EquipConfiguration::ZoneName );
 		ReturnAirNodeNumber = 0; // default is not found
 		if ( ControlledZoneIndex > 0 ) {
+			{ auto const & thisZoneEquip( ZoneEquipConfig( ControlledZoneIndex ) );
+			if ( thisZoneEquip.ActualZoneNum > 0 ) {
+				if ( NodeName == "" ) {
+					// If NodeName is blank, return first return node number, but warn if there are multiple return nodes for this zone
+					ReturnAirNodeNumber = thisZoneEquip.ReturnNode( 1 );
+					if ( thisZoneEquip.NumReturnNodes > 1 ){ 
+						ShowWarningError( "GetReturnAirNodeForZone: " + calledFromDescription + ", request for zone return node is ambiguous." );
+						ShowContinueError( "Zone=" + thisZoneEquip.ZoneName + " has "+ General::RoundSigDigits(thisZoneEquip.NumReturnNodes ) + " return nodes. First return node will be used." );
+					}
+				} else {
+					for ( int nodeCount = 1; nodeCount <= thisZoneEquip.NumReturnNodes; ++nodeCount ) {
+						int curNodeNum = thisZoneEquip.ReturnNode( nodeCount );
+						if ( NodeName == DataLoopNode::NodeID( curNodeNum ) ) {
+							ReturnAirNodeNumber = curNodeNum;
+						}
+					}
+				}
+			}}
+		}
+
+		return ReturnAirNodeNumber;
```

Comment:
So, at some point, when the entire codebase is setup to handle the possibility of multiple nodes per zone, this function should never need to warn anymore, right?  The NodeName will be required and provided once the callers all understand the new paradigm.  I hope so, but if I'm wrong then I just want to understand why.

Reply:
Good question.  I had wanted the node names to be optional in places like return air heat gain from lights so if you do have only one return node, you don't have to enter a node name.  If we make the return node required, then it wouldn't work with a zonelist name anymore.  So, not sure.

Reply:
That is interesting.  And tricky.  I see your point, but it feels funny as optional.  Nothing to worry about for today, that's for sure.  We'll see how it plays out.